instrumentation/opentelemetry_redix/test/opentelemetry_redix_test.exs

```diff
@@ -0,0 +1,88 @@
+defmodule OpentelemetryRedixTest do
+  use ExUnit.Case, async: false
+
+  doctest OpentelemetryRedix
+
+  require OpenTelemetry.Tracer
+  require OpenTelemetry.Span
+  require Record
+
+  for {name, spec} <- Record.extract_all(from_lib: "opentelemetry/include/otel_span.hrl") do
+    Record.defrecord(name, spec)
+  end
+
+  for {name, spec} <- Record.extract_all(from_lib: "opentelemetry_api/include/opentelemetry.hrl") do
+    Record.defrecord(name, spec)
+  end
+
+  setup do
+    :application.stop(:opentelemetry)
+    :application.set_env(:opentelemetry, :tracer, :otel_tracer_default)
+
+    :application.set_env(:opentelemetry, :processors, [
+      {:otel_batch_processor, %{scheduled_delay_ms: 1}}
+    ])
+
+    :application.start(:opentelemetry)
+
+    :otel_batch_processor.set_exporter(:otel_exporter_pid, self())
+
+    OpenTelemetry.Tracer.start_span("test")
+
+    on_exit(fn ->
+      OpenTelemetry.Tracer.end_span()
+    end)
+  end
+
+  test "records span on commands" do
+    OpentelemetryRedix.setup()
+
+    conn = start_supervised!({Redix, []})
+
+    {:ok, "OK"} = Redix.command(conn, ["SET", "foo", "bar"])
+
+    assert_receive {:span,
+                    span(
+                      name: "SET",
+                      kind: :client,
+                      attributes: list
+                    )}
+
+    assert [
+             "db.operation": "SET",
+             "db.statement": "SET foo ?",
+             "db.system": "redis",
+             "net.peer.name": "localhost",
+             "net.peer.port": "6379"
+           ] = List.keysort(list, 0)
+  end
+
+  test "records span on piplines" do
+    OpentelemetryRedix.setup()
+
+    conn = start_supervised!({Redix, []})
+
+    {:ok, [_, _, _, "2"]} =
+      Redix.pipeline(conn, [
+        ["DEL", "counter"],
+        ["INCR", "counter"],
+        ["INCR", "counter"],
+        ["GET", "counter"]
+      ])
+
+    assert_receive {:span,
+                    span(
+                      name: "pipeline",
+                      kind: :client,
+                      attributes: list
+                    )}
+
+    assert [
+             "db.operation": "pipeline",
+             "db.statement": "DEL counter; INCR counter; INCR counter; GET counter",
```

Comment:
The failing test appears to be that this is emitted with new lines between commands.

Reply:
🙈 updated the implementation after looking at some results, forgot to update the tests (and check here). Sorry for that!